modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java

```diff
@@ -0,0 +1,124 @@
+
+    @Factory(dataProvider = "restAPIUserConfigProvider")
+    public UserStoreFailureTest(TestUserMode userMode) throws Exception {
+
+        super.init(userMode);
+        this.context = isServer;
+        this.authenticatingUserName = context.getContextTenant().getTenantAdmin().getUserName();
+        this.authenticatingCredential = context.getContextTenant().getTenantAdmin().getPassword();
+        this.tenant = context.getContextTenant().getDomain();
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+
+        super.testInit(API_VERSION, swaggerDefinition, tenant);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testConclude() {
+
+        super.conclude();
+    }
+
+    @BeforeMethod(alwaysRun = true)
+    public void testInit() {
+
+        RestAssured.basePath = basePath;
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void testFinish() {
+
+        RestAssured.basePath = StringUtils.EMPTY;
+    }
+
+    @DataProvider(name = "restAPIUserConfigProvider")
+    public static Object[][] restAPIUserConfigProvider() {
+
+        return new Object[][]{
+                {TestUserMode.SUPER_TENANT_ADMIN},
+                {TestUserMode.TENANT_ADMIN}
+        };
+    }
+
+    @Test
+    public void testAddUserStoreNegativeCase() throws Exception {
+
+        String body = readResource("add-secondary-user-store-negative.json");
+        Response response = getResponseOfPost(USER_STORE_PATH_COMPONENT, body);
+        validateErrorResponse(response, HttpStatus.SC_INTERNAL_SERVER_ERROR, "SUS-50001");
+    }
+
+    @Test
+    public void testGetUserStoreByDomainIdNegativeCase() {
+
+        Response response = getResponseOfGet(USER_STORE_PATH_COMPONENT + PATH_SEPARATOR +
+                INCORRECT_DOMAIN_ID);
+        validateErrorResponse(response, HttpStatus.SC_NOT_FOUND, "SUS-50012");
+    }
+
+    @Test
+    public void testUpdateUserStoreByIdNegativeCase() throws IOException {
+
+        String body = readResource("add-secondary-user-store-negative.json");
+        Response response = getResponseOfPut(USER_STORE_PATH_COMPONENT + PATH_SEPARATOR +
+                INCORRECT_DOMAIN_ID, body);
+        validateErrorResponse(response, HttpStatus.SC_INTERNAL_SERVER_ERROR, "SUS-50004");
+    }
+
+    @Test
+    public void testGetUserStoreTypeByIdNegativeCase() {
+
+        Response response = getResponseOfGet(USER_STORE_PATH_COMPONENT + USER_STORE_META_COMPONENT
+                + PATH_SEPARATOR + INCORRECT_TYPE_ID);
+        response.then().log().ifValidationFails().assertThat().statusCode(HttpStatus.SC_OK)
+                .body("", Matchers.hasSize(0));
+    }
+
+    @Test
+    public void testDeleteUserStoreByIdNegativeCase() {
+
+        Response response = getResponseOfDelete(USER_STORE_PATH_COMPONENT + PATH_SEPARATOR
+                + "SkRCQy0DRETYUI");
+        validateErrorResponse(response, HttpStatus.SC_INTERNAL_SERVER_ERROR, "SUS-50002");
```

Comment:
This should not have returned internal server error. If the domain is incorrect, it should return not found. Fix the endpoint side to do so.

Reply:
Fixed in a50eaccbf9929d07a1c6542b920d4add8bed472a